CDL-Dryad/dryad-app
# New Merritt Deposit API

https://github.com/CDL-Dryad/dryad-product-roadmap/issues/2068

This removes SWORD from the deposit process and instead uses the Merritt deposit API which is always asynchronous.  (It should return acceptance right away but full processing doesn't happen for a while.)

You'll notice I removed a whole lot of classes and re-integrated this as `stash-deposit` gem instead of the `stash-sword` one.  I tried to clean the basic gem up and make it a lot simpler.

I still think there are some questions about the whole pluggable gem design for all these things which is a bit overkill for the way we actually use things and adds a lot of extra complexity.  However, I simplified this specific gem a lot without doing a giant rework of all the Merritt, queuing, promises and other pieces of code that do other parts.

I also left some of the configuration things alone (like the endpoint config).  See https://github.com/CDL-Dryad/dryad-product-roadmap/issues/2332 .  I think this should make it easier to revert to SWORD if something is catastrophically bad about the new Merritt endpoints.  Later we can convert SWORD endpoints to just the collection names.

To test:

- Add a dataset and submit it.  Ensure it goes through Merritt correctly.
- Update a dataset and submit updates.  Ensure updates go through and a new version is created.
- This shouldn't change much about our current queuing setup which we can also rework later since there may be less reason to continue using it when all Merritt submissions are now asynchronous.

I'll check with the Merritt people to see what they'd like to do for testing and if release on Tuesday is feasible. I'll mark this a draft PR until then.
